Document/0x04a-Android.md

```diff
@@ -189,6 +189,121 @@ drwxr-xr-x    9 sven  staff   306B Dec  5 16:29 smali
+    #define AID_TRUNKS        1049  /* trunksd process (TPM daemon) */
+    #define AID_NVRAM         1050  /* Access-controlled NVRAM */
+    #define AID_DNS           1051  /* DNS resolution daemon (system: netd) */
+    #define AID_DNS_TETHER    1052  /* DNS resolution daemon (tether: dnsmasq) */
+    /* Changes to this file must be made in AOSP, *not* in internal branches. */
+
+    #define AID_SHELL         2000  /* adb and debug shell user */
+    #define AID_CACHE         2001  /* cache access */
+    #define AID_DIAG          2002  /* access to diagnostic resources */
+
+    /* The range 2900-2999 is reserved for OEM, and must never be
+     * used here */
+    #define AID_OEM_RESERVED_START 2900
+    #define AID_OEM_RESERVED_END   2999
+
+    /* The 3000 series are intended for use as supplemental group id's only.
+     * They indicate special Android capabilities that the kernel is aware of. */
+    #define AID_NET_BT_ADMIN  3001  /* bluetooth: create any socket */
+    #define AID_NET_BT        3002  /* bluetooth: create sco, rfcomm or l2cap sockets */
+    #define AID_INET          3003  /* can create AF_INET and AF_INET6 sockets */
+    #define AID_NET_RAW       3004  /* can create raw INET sockets */
+    #define AID_NET_ADMIN     3005  /* can configure interfaces and routing tables. */
+    #define AID_NET_BW_STATS  3006  /* read bandwidth statistics */
+    #define AID_NET_BW_ACCT   3007  /* change bandwidth statistics accounting */
+    #define AID_NET_BT_STACK  3008  /* bluetooth: access config files */
+    #define AID_READPROC      3009  /* Allow /proc read access */
+    #define AID_WAKELOCK      3010  /* Allow system wakelock read/write access */
+
+    /* The range 5000-5999 is also reserved for OEM, and must never be used here. */
+    #define AID_OEM_RESERVED_2_START 5000
+    #define AID_OEM_RESERVED_2_END   5999
+
+    #define AID_EVERYBODY     9997  /* shared between all apps in the same profile */
+    #define AID_MISC          9998  /* access to misc storage */
+    #define AID_NOBODY        9999
+
+    #define AID_APP          10000  /* first app user */
+
+    #define AID_ISOLATED_START 99000 /* start of uids for fully isolated sandboxed processes */
+    #define AID_ISOLATED_END   99999 /* end of uids for fully isolated sandboxed processes */
+
+    #define AID_USER        100000  /* offset for uid ranges for each user */
+
+    #define AID_SHARED_GID_START 50000 /* start of gids for apps in each user to share */
+    #define AID_SHARED_GID_END   59999 /* start of gids for apps in each user to share */
+```
+
+What is shown when the `id` command is executed on an Android shell is the user and group ids applied for the current process.
+```
+$ adb shell id
+uid=2000(shell) gid=2000(shell) groups=2000(shell),1004(input),1007(log),1011(adb),1015(sdcard_rw),1028(sdcard_r),3001(net_bt_admin),3002(net_bt),3003(inet),3006(net_bw_stats),3009(readproc) context=u:r:shell:s0
+```
```

Comment:
Can you also add a paragraph that for each installed App a new user will be created and only this users has access to the data directory of the app etc?